src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -336,44 +930,81 @@ func (s *commitLogSource) shouldIncludeInIndex(
 	return rangesToBootstrap.Overlaps(indexBlockRange)
 }
 
-func (s *commitLogSource) mergeShards(
+func (s *commitLogSource) mergeAllShardsCommitLogEncodersAndSnapshots(
+	ns namespace.Metadata,
+	shardsTimeRanges result.ShardTimeRanges,
+	snapshotFiles map[uint32]fs.FileSetFilesSlice,
+	mostRecentCompleteSnapshotByBlockShard map[xtime.UnixNano]map[uint32]snapshotMetadata,
 	numShards int,
-	bopts result.Options,
 	blockSize time.Duration,
-	blopts block.Options,
-	encoderPool encoding.EncoderPool,
 	unmerged []shardData,
-) result.DataBootstrapResult {
+) (result.DataBootstrapResult, error) {
 	var (
-		shardErrs               = make([]int, numShards)
-		shardEmptyErrs          = make([]int, numShards)
-		bootstrapResult         = result.NewDataBootstrapResult()
-		blocksPool              = bopts.DatabaseBlockOptions().DatabaseBlockPool()
-		multiReaderIteratorPool = blopts.MultiReaderIteratorPool()
+		shardErrs       = make([]int, numShards)
+		shardEmptyErrs  = make([]int, numShards)
+		bootstrapResult = result.NewDataBootstrapResult()
 		// Controls how many shards can be merged in parallel
 		workerPool          = xsync.NewWorkerPool(s.opts.MergeShardsConcurrency())
 		bootstrapResultLock sync.Mutex
 		wg                  sync.WaitGroup
 	)
-
 	workerPool.Init()
 
 	for shard, unmergedShard := range unmerged {
+		snapshotData, err := s.bootstrapShardSnapshots(
```

Comment:
We could possibly read the snapshots in parallel too yeah?  We could just add this to the worker func below which we spawn off?